loader/src/sources/vts/geo.js

```diff
@@ -0,0 +1,164 @@
+const got = require("got");
+const Sentry = require("@sentry/node");
+
+const dataProviders = {
+  "Rite-Aid": {
+    provider: "rite_aid",
+    getStoreName: (data) => {
+      const m = data.name.match(/RITE AID PHARMACY #?(\d+)/i);
+      if (m) {
+        return `Rite Aid #${parseInt(m[1], 10)}`; // format as in riteaid/api.js
+      }
+      return data.name;
+    },
+  },
+  CVS: {
+    // modeled after code in cvs/api.js
+    provider: "cvs",
+    getStoreName: (data) => {
+      const m = data.name.match(/CVS Pharmacy, Inc. #?(\d+)/i);
+      if (m) {
+        return `CVS #${parseInt(m[1], 10)}`; // format as in cvs/api.js
+      }
+      return data.name;
+    },
+  },
+};
+
+function warn(message, context) {
+  console.warn(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Info);
+}
+
+function error(message, context) {
+  console.error(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Error);
+}
+
+async function getStores() {
+  try {
+    const response = await got({
+      url: `https://univaf-data-snapshots.s3.us-west-2.amazonaws.com/vts/vts-final-output-locations.geojson`,
+    });
+    return JSON.parse(response.body);
+  } catch (e) {
+    error(`Error fetching stored Vaccine the States data`, e);
+    return [];
+  }
+}
+
+/**
+ * Filter out some stores with bad data.
+ * @param {any} store
+ * @returns {boolean}
+ */
+function hasUsefulData(store) {
+  return (
+    store.properties.concordances?.length &&
+    store.geometry?.coordinates &&
+    store.properties.provider?.name &&
+    store.properties.provider?.name in dataProviders
+  );
+}
+
+function formatStore(store) {
+  const data = store.properties;
+
+  let result;
+  Sentry.withScope((scope) => {
+    const dataProvider = dataProviders[data.provider.name];
+    if (!dataProvider) {
+      error(`Unexpected provider name ${data.provider.name}`);
+      return null;
+    }
+
+    const provider = dataProvider.provider;
+    const name = dataProvider.getStoreName(data);
+
+    scope.setContext("location", {
+      id: store.id,
+      url: data.vts_url,
+      name: name,
+      provider: provider,
+    });
+
+    function splitConcordance(concordance) {
+      const colon = concordance.indexOf(":");
+      return [
+        concordance.substring(0, colon),
+        concordance.substring(colon + 1),
+      ];
+    }
+
+    function filterConcordancePairs(pairs) {
+      return pairs
+        .filter(([systemCode, value]) => {
+          const system = systemsToSend[systemCode];
```

Comment:
It looks funny, but does not matter since the name isn't looked up until later when it's called. Arguably there's too much all in this area, but I was trying to do it within the `Sentry.withScope` and `scope.setContext`.